### ISSUE TITLE
[rllib] Bug of AutoRegressiveDistribution: the old policy still can have access to current model parameters when using PPO

### ISSUE THREAD

**Issue by yangysc** [ray-project/ray#17108]
### What is the problem?
For example, when initializing a **TorchBinaryAutoregressiveDistribution** instance, the `_a1_distribution` and `_a2_distribution` need to call `self.model.action_module`, which is the model parameters defined in policy network.
https://github.com/ray-project/ray/blob/86d0159c0af5cc613cf91005d61edff8b761f84b/rllib/examples/models/autoregressive_action_dist.py#L132-L143

When using PPO, we need to calculate the difference between old policy and current policy,
https://github.com/ray-project/ray/blob/ac54164e73cf554ea51452ba3fb4ece4f7017623/rllib/agents/ppo/ppo_torch_policy.py#L45-L46
https://github.com/ray-project/ray/blob/ac54164e73cf554ea51452ba3fb4ece4f7017623/rllib/agents/ppo/ppo_torch_policy.py#L67-L68
https://github.com/ray-project/ray/blob/ac54164e73cf554ea51452ba3fb4ece4f7017623/rllib/agents/ppo/ppo_torch_policy.py#L73
And the possible problem is the `prev_action_dist ` **uses the current model instead of the old policy**. So the `kl` between two policies is not accurate.

- **How can we backup an old model before the current training  `K` epoches?**
- A another solution is we save all logits into SampleBatch and use them in calculating PPO loss.
- A another solution is to approximate the kl between two distributions using sampled actions
 `kl(q, p) = r-1 - log r` and `r = p(x)/q(x)`

*Ray version and other system information (Python version, TensorFlow version, OS):*
- Python 3.8
- Pytorch 1.8
- Ubuntu
### Reproduction (REQUIRED)
Please provide a short code snippet (less than 50 lines if possible) that can be copy-pasted to reproduce the issue. The snippet should have **no external library dependencies** (i.e., use fake or mock data / environments):

If the code snippet cannot be run by itself, the issue will be closed with "needs-repro-script".

- [X] I have verified my script runs in a clean environment and reproduces the issue.
- [X] I have verified the issue also occurs with the [latest wheels](https://docs.ray.io/en/master/installation.html).

**Comment by avnishn** [comment]
Saving the logits seems like it wouldn't be a fix, since the logits are conditioned on technically, old observations. This is fine in the case of PPO, but it would not work for any other case AFAICT. I'll investigate further.
